site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -2716,10 +2716,20 @@ else if (state.getAttribute(STATE_CM_REQUESTED_SECTIONS) != null) {
 					context.put("publicChangeable", Boolean.FALSE);
 				}
 				context.put("include", Boolean.valueOf(siteInfo.getInclude()));
-				if(Instant.now().isAfter(new Date(courseManagementService.getAcademicSession(siteInfo.term).getStartDate().getTime()).toInstant()) && Instant.now().isBefore(new Date(courseManagementService.getAcademicSession(siteInfo.term).getEndDate().getTime()).toInstant())){	//set to Published by default if we're in the site's term
-					context.put("published", true);
+
```

Comment:
What about this?

boolean sitePublished = true;
// If this site is a course site, publish if we're inside the term dates
if (siteInfo.site_type != null && SiteTypeUtil.isCourseSite(siteInfo.site_type)) {
	AcademicSession academicSession  = courseManagementService.getAcademicSession(siteInfo.term);
        // I don't like this but I understand it was accepted in the previous PR, if that comes from my team it will surely have 15 comments.
        sitePublished = Instant.now().isAfter(new Date(academicSession.getStartDate().getTime()).toInstant())
						&& Instant.now().isBefore(new Date(academicSession.getEndDate().getTime()).toInstant());
}
context.put("published", sitePublished);